argilla-server/docker/argilla-hf-spaces/scripts/start.sh

```diff
@@ -8,8 +8,11 @@ export OAUTH2_HUGGINGFACE_CLIENT_ID=$OAUTH_CLIENT_ID
 export OAUTH2_HUGGINGFACE_CLIENT_SECRET=$OAUTH_CLIENT_SECRET
 export OAUTH2_HUGGINGFACE_SCOPE=$OAUTH_SCOPES
 
-# Set the space author name as username if no provided.
+# Set the space creator name as username if no name is provided, if the user is not found, use the provided space author name
 # See https://huggingface.co/docs/hub/en/spaces-overview#helper-environment-variables for more details
-export USERNAME="${USERNAME:-$SPACE_AUTHOR_NAME}"
+DEFAULT_USERNAME=$(curl -L -s https://huggingface.co/api/users/${SPACES_CREATOR_USER_ID}/overview | jq -r '.user' || echo "${SPACE_AUTHOR_NAME}")
+export USERNAME="${USERNAME:-$DEFAULT_USERNAME}"
+DEFAULT_PASSWORD=$(pwgen -s 16 1)
+export PASSWORD="${PASSWORD:-$DEFAULT_PASSWORD}"
```

Comment:
I don't think this has implications except for always creating a user with a random password (this will be printed in the logs), if no password is provided. Otherwise, no explicit user will be created, which feels weird to me in general.

Reply:
@frascuchon I updated the docs for spaces specifically because it only makes sense there.